### PR TITLE
Cleaned Up API Commands

### PR DESCRIPTION
**Pull request by openbots-ff**
Cleaned up files under the API Commands directory. Moved GetHTML and QueryHTML to Web Browser Commands

### CODE REVIEW

File: taskt/Core/Automation/Commands/API Commands/ExecuteRESTAPICommand.cs [hunk 1]
```diff
@@ -1,356 +1,333 @@
...
     {
         [XmlAttribute]
-        [Attributes.PropertyAttributes.PropertyDescription("Please enter the Base URL (ex. http://mysite.com)")]
-        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
-        [Attributes.PropertyAttributes.InputSpecification("Define any API endpoint which contains the full URL.")]
-        [Attributes.PropertyAttributes.SampleUsage("**https://example.com** or **{vMyUrl}**")]
-        [Attributes.PropertyAttributes.Remarks("")]
+        [PropertyDescription("Base URL")]
+        [InputSpecification("Provide the base URL of the API.")]
+        [SampleUsage("https://example.com or {vMyUrl}")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/API Commands/ExecuteRESTAPICommand.cs]: Please use || (double pipes) to separate two or more sample usages.

Comment (openbots-ff) [reply taskt/Core/Automation/Commands/API Commands/ExecuteRESTAPICommand.cs]: Thanks. Forgot those. Done.

File: taskt/Core/Automation/Commands/API Commands/ExecuteRESTAPICommand.cs [hunk 2]
```diff
@@ -1,356 +1,333 @@
...
 
         [XmlAttribute]
-        [Attributes.PropertyAttributes.PropertyDescription("Please enter the endpoint (Ex. /v2/endpoint)")]
-        [Attributes.PropertyAttributes.PropertyUIHelper(Attributes.PropertyAttributes.PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
-        [Attributes.PropertyAttributes.InputSpecification("Define any API endpoint which contains the full URL.")]
-        [Attributes.PropertyAttributes.SampleUsage("**/v2/getUser/1** or **{vMyUrl}**")]
-        [Attributes.PropertyAttributes.Remarks("")]
+        [PropertyDescription("Endpoint")]
+        [InputSpecification("Define any API endpoint which contains the full URL.")]
+        [SampleUsage("/v2/getUser/1 or {vMyUrl}")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/API Commands/ExecuteRESTAPICommand.cs]: Use || (Double Pipes) for separation for all references.

File: taskt/Core/Automation/Commands/Web Browser Commands/GetHTMLSourceCommand.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        public string v_WebRequestURL { get; set; }
+
+        [XmlAttribute]
+        [PropertyDescription("Execute Request As Logged On User")]
+        [PropertyUISelectionOption("Yes")]
+        [PropertyUISelectionOption("No")]
+        [InputSpecification("Sets currently logged on user authentication information for the request.")]
+        [SampleUsage("")]
+        [Remarks("")]
+        [PropertyUIHelper(PropertyUIHelper.UIAdditionalHelperType.ShowVariableHelper)]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Web Browser Commands/GetHTMLSourceCommand.cs]: We might not need the UIHelper for the drop down.
Please remove.

File: taskt/Core/Automation/Commands/Web Browser Commands/QueryHTMLSourceCommand.cs [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    [Serializable]
+    [Group("Web Browser Commands")]
+    [Description("This command parses and extracts data from an HTML source object or a successful **GetHTMLSourceCommand**.")]
+
+    public class QueryHTMLSourceCommand : ScriptCommand
+    {
+        [XmlAttribute]
+        [PropertyDescription("HTML")]
+        [InputSpecification("Enter the HTML to be queried.")]
+        [SampleUsage("<!DOCTYPE html><html><head><title>Example</title></head></html> or {vMyHTML}")]
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Web Browser Commands/QueryHTMLSourceCommand.cs]: Use || (Double Pipes) for separation of Sample Usages throughout this file.

File: taskt/Core/Automation/Commands/Web Browser Commands/QueryHTMLSourceCommand.cs [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        {
+            HtmlAgilityPack.HtmlDocument doc = new HtmlAgilityPack.HtmlDocument();
+            doc.LoadHtml(v_HTMLVariable.ConvertToUserVariable(sender));
+
+            var div = doc.DocumentNode.SelectSingleNode(v_XPathQuery);
+            string divString = div.InnerText;
+
+            divString.StoreInUserVariable(sender, v_OutputUserVariableName);
+
+
```
Comment (ob-ab) [inline taskt/Core/Automation/Commands/Web Browser Commands/QueryHTMLSourceCommand.cs]: Please remove Extra Lines

**Review by ob-ab — COMMENTED** [review]
Please make a few small changes and everything else looks good to me.

File: taskt/Core/Automation/Commands/Web Browser Commands/GetHTMLSourceCommand.cs (hunk 3 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Web Browser Commands/GetHTMLSourceCommand.cs]: good catch. done

File: taskt/Core/Automation/Commands/Web Browser Commands/QueryHTMLSourceCommand.cs (hunk 5 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/Web Browser Commands/QueryHTMLSourceCommand.cs]: done

**Review by ob-ab — DISMISSED** [review]
Looks good to me.

File: taskt/Core/Automation/Commands/API Commands/ExecuteDLLCommand.cs [hunk 6]
```diff
@@ -1,86 +1,91 @@
...
 namespace taskt.Core.Automation.Commands
 {
 
     [Serializable]
-    [Attributes.ClassAttributes.Group("API Commands")]
-    [Attributes.ClassAttributes.Description("This command allows user to invoke a method of a specific class from DLL")]
-    [Attributes.ClassAttributes.UsesDescription("Use this command to invoke a specific method of a class from DLL (Library) File")]
-    [Attributes.ClassAttributes.ImplementationDescription("TBD")]
+    [Group("API Commands")]
+    [Description("This command allows user to invoke a method of a specific class from a DLL.")]
```
Comment (CameronHerwig) [inline taskt/Core/Automation/Commands/API Commands/ExecuteDLLCommand.cs]: "allows user" is a bit awkward wording. Might want to put "allows users" or "allows the user"

Comment (CameronHerwig) [reply taskt/Core/Automation/Commands/API Commands/ExecuteDLLCommand.cs]: The other commands appear to use "allows you".

File: taskt/Core/Automation/Commands/API Commands/ExecuteRESTAPICommand.cs [hunk 7]
```diff
@@ -1,356 +1,333 @@
...
 namespace taskt.Core.Automation.Commands
 {
 
     [Serializable]
-    [Attributes.ClassAttributes.Group("API Commands")]
-    [Attributes.ClassAttributes.Description("This command allows user to call REST API.")]
-    [Attributes.ClassAttributes.UsesDescription("Use this command when you want to call a REST API with specific HTTP method")]
-    [Attributes.ClassAttributes.ImplementationDescription("This command implements 'RestRequest' to call a REST API Endpoint with specific HTTP 'Method'")]
+    [Group("API Commands")]
+    [Description("This command allows user to call a REST API with a specific HTTP method.")]
```
Comment (CameronHerwig) [inline taskt/Core/Automation/Commands/API Commands/ExecuteRESTAPICommand.cs]: Same as above. "allows you"

File: taskt/Core/Automation/Commands/API Commands/ExecuteDLLCommand.cs (hunk 6 above)
Comment (openbots-ff) [reply taskt/Core/Automation/Commands/API Commands/ExecuteDLLCommand.cs]: Yeah, pretty sure that was just a typo. I'll fix it. I don't really like the "allows users" part at all so I've been removing it from all the excel commands. Done.

**Review by CameronHerwig — APPROVED** [review]

**Review by ob-ab — APPROVED** [review]

**Pull request merged**